hummingbot/connector/exchange/binance/binance_exchange.pyx

```diff
@@ -460,14 +460,13 @@ cdef class BinanceExchange(ExchangeBase):
                 continue
             for trade in trades:
                 if self.is_confirmed_new_order_filled_event(str(trade["id"]), str(trade["orderId"]), trading_pair):
-                    client_order_id = get_client_order_id("buy" if trade["isBuyer"] else "sell", trading_pair)
                     # Should check if this is a partial filling of a in_flight order.
                     # In that case, user_stream or _update_order_fills_from_trades will take care when fully filled.
-                    if client_order_id not in self.in_flight_orders:
+                    if not any(trade["id"] in in_flight_order.trade_id_set for in_flight_order in self._in_flight_orders.values()):
                         self.c_trigger_event(self.MARKET_ORDER_FILLED_EVENT_TAG,
                                              OrderFilledEvent(
                                                  trade["time"],
-                                                 client_order_id,
+                                                 get_client_order_id("buy" if trade["isBuyer"] else "sell", trading_pair),
```

Comment:
I don't think we should generate a new client id at this point. We use the client id to identify orders that were created by the bot. Order created manually shouldn't need a client id attached to them. Also, orders created by the bot should have same client id that was generated when the order was created by the bot. 
Lastly, I think we should use https://binance-docs.github.io/apidocs/spot/en/#all-orders-user_data endpoint for history reconciliation as it include bot client and exchange order id.
Then

Reply:
I've tried using that endpoint instead of the currently being used (myTrades) but unfortunately output info lacks the commission paid in the trade. That's important to create the order filled event. I couldn't find an way to get the commission other than calling other endpoints too.
![image](https://user-images.githubusercontent.com/14949838/106678166-e832a680-6598-11eb-8588-941d3b3cc967.png)

On the other hand, think it's a good idea and will implement your suggestion to reuse the original local order id instead of generating a new one. This requires further changes which will be proposed in this PR.